scheme/sevsnp/evidence_handler.go

```diff
@@ -0,0 +1,434 @@
+	}
+
+	err := json.Unmarshal(taEndorsement.Attributes, &avk)
+	if err != nil {
+		return err
+	}
+
+	provisionedArk := avk.VerifKeys[0]
+
+	// Parse certificate chain in evidence (auxblob)
+	err = tsm.FromCBOR(token.Data)
+	if err != nil {
+		return err
+	}
+
+	protoReport, err := abi.ReportToProto(tsm.OutBlob)
+	if err != nil {
+		return err
+	}
+
+	ark, err := getARK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	arkBlock, err := readCert(ark)
+	if err != nil {
+		return err
+	}
+
+	ask, err := getASK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	askBlock, err := readCert(ask)
+	if err != nil {
+		return err
+	}
+
+	vcek, err := getVCEK(tsm.AuxBlob)
+	if err != nil {
+		return err
+	}
+
+	vcekBlock, err := readCert(vcek)
+	if err != nil {
+		return err
+	}
+
+	// Test if TA matches with the one supplied in evidence
+	match, err := compareTAs([]byte(provisionedArk.String()), ark)
+	if err != nil {
+		return err
+	}
+	if !match {
+		return fmt.Errorf("ARK in evidence does not match provisioned ARK")
+	}
+
+	// Validate the integrity of evidence by ensuring the
+	// certificate chain is intact, and the signature is valid
+	var attestation sevsnp.Attestation
+	attestation.Report = protoReport
+	attestation.CertificateChain = &sevsnp.CertificateChain{VcekCert: vcekBlock, AskCert: askBlock, ArkCert: arkBlock}
+	err = verify.SnpAttestation(&attestation, snpAttestationOptions())
+
+	if err != nil {
+		log.Errorf("failed to validate certificate chain: %+v\n", err)
+	}
+
+	return err
+}
+
+// refvalToComidTriple converts extracted reference values to CoMID value triple
+func refvalToComidTriple(endorsementsStrings []string) (*comid.ValueTriple, error) {
+	var (
+		refValEndorsement handler.Endorsement
+		rv                comid.ValueTriple
+	)
+
+	for i, e := range endorsementsStrings {
+		var endorsement handler.Endorsement
+
+		if err := json.Unmarshal([]byte(e), &endorsement); err != nil {
+			return nil, fmt.Errorf("could not decode endorsement at index %d: %w", i, err)
+		}
+
+		if endorsement.Type == handler.EndorsementType_REFERENCE_VALUE {
+			refValEndorsement = endorsement
+			break
+		}
+	}
+
+	if refValEndorsement.Type != handler.EndorsementType_REFERENCE_VALUE {
+		return nil, fmt.Errorf("reference values unavailable")
+	}
+
+	err := json.Unmarshal(refValEndorsement.Attributes, &rv)
+	if err != nil {
+		return nil, err
+	}
+
+	return &rv, nil
+}
+
+// evidenceToComidTriple converts claim set to CoMID value triple
+func evidenceToComidTriple(ec *proto.EvidenceContext) (*comid.ValueTriple, error) {
+	evCorimJson, err := json.Marshal(ec.Evidence.AsMap())
+	if err != nil {
+		return nil, err
+	}
+
+	evComid, err := comidFromJson(evCorimJson)
+	if err != nil {
+		return nil, err
+	}
+
+	return &evComid.Triples.ReferenceValues.Values[0], nil
+}
+
+// compareMeasurements checks if two given comid.Measurement variables are the same.
+func compareMeasurements(refM comid.Measurement, evM comid.Measurement) bool {
```

Comment:
This seems like a generic function for the base spec's measurement-map. Shouldn't this be something we can import from a generic library for appraisal against a base CoRIM?

Reply:
That makes sense. It's not specific to SEV-SNP per se, it will benefit from being in a library. Let me do that change.